core/src/sync/message/checkpoint_blame_state.rs

```diff
@@ -0,0 +1,166 @@
+// Copyright 2019 Conflux Foundation. All rights reserved.
+// Conflux is free software and distributed under GNU General Public License.
+// See http://www.gnu.org/licenses/
+
+use crate::{
+    message::{HasRequestId, Message, MsgId, RequestId},
+    sync::{
+        message::{msgid, Context, Handleable, KeyContainer},
+        request_manager::Request,
+        Error, ErrorKind, ProtocolConfiguration,
+    },
+};
+use cfx_types::H256;
+use rlp::{Decodable, DecoderError, Encodable, Rlp, RlpStream};
+use std::{any::Any, time::Duration};
+
+#[derive(Debug, Clone)]
+pub struct CheckpointBlameStateRequest {
+    pub request_id: u64,
+    pub trusted_blame_block: H256,
+}
+
+impl CheckpointBlameStateRequest {
+    pub fn new(trusted_blame_block: H256) -> Self {
+        Self {
+            request_id: 0,
+            trusted_blame_block,
+        }
+    }
+}
+
+build_msg_impl! { CheckpointBlameStateRequest, msgid::GET_CHECKPOINT_BLAME_STATE_REQUEST, "CheckpointBlameStateRequest" }
+build_has_request_id_impl! { CheckpointBlameStateRequest }
+
+impl Request for CheckpointBlameStateRequest {
+    fn as_message(&self) -> &Message { self }
+
+    fn as_any(&self) -> &Any { self }
+
+    fn timeout(&self, conf: &ProtocolConfiguration) -> Duration {
+        conf.blocks_request_timeout
+    }
+
+    fn on_removed(&self, _inflight_keys: &mut KeyContainer) {}
+
+    fn with_inflight(&mut self, _inflight_keys: &mut KeyContainer) {}
+
+    fn is_empty(&self) -> bool { false }
+
+    fn resend(&self) -> Option<Box<Request>> { Some(Box::new(self.clone())) }
+}
+
+impl Handleable for CheckpointBlameStateRequest {
+    fn handle(self, ctx: &Context) -> Result<(), Error> {
+        let trusted_block = ctx
+            .manager
+            .graph
+            .data_man
+            .block_header_by_hash(&self.trusted_blame_block);
+        if trusted_block.is_none() {
+            warn!(
+                "invalid trusted_blame_block={} in CheckpointBlameStateRequest from peer={}",
+                self.trusted_blame_block,
+                ctx.peer,
+            );
+            return Err(ErrorKind::Invalid.into());
```

Comment:
For a node in catch-up state, it's better to respond something instead of return error and disconnect.

And the request sender has to wait for timeout to resend the request again.

Please also double confirm similar error handling in this method.